### PR TITLE
[msbuild] Add nuget diagnostics.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by akoeplinger** [comment]
Had to abort it as it caused 400 MB of log spam. Not sure why yet.

**Comment by akoeplinger** [comment]
Hmm, no idea what's up, it has a million empty newlines in the logfile when it got to the msbuild step...
I retriggered it, let's see what happens.

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by akoeplinger** [comment]
Hmm, same issue :( Can you please try removing the `-Verbosity detailed` from restore?

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by akoeplinger** [comment]
Very weird, seems to be an issue with nuget sources list, though I don't see this behavior if I execute it on the bot over SSH....

Looks like we need to fallback to `cat ~/.config/NuGet/NuGet.Config`

File: msbuild/Makefile [hunk 1]
```diff
@@ -333,6 +333,7 @@ $(MAC_DESTDIR)$(MAC_FRAMEWORK_CURRENT_DIR)/lib/msbuild/%.dll: build/%.dll | $(MA
 all-local:: $(MSBUILD_PRODUCTS) test-xml
 
 .build-stamp: $(ALL_SOURCES)
+	cat ~/.config/NuGet/NuGet.Config
```
Comment (akoeplinger) [inline msbuild/Makefile]: The file might not exist on some systems, so probably better to do `|| true` so it doesn't fail the build. Sorry, should've mentioned this right away.

**Comment by xamarin-release-manager** [comment]
Build success

**Pull request merged**